### ISSUE TITLE
about attention_decoder.py

### ISSUE THREAD

**Issue by jimmyljxy** [yaserkl/RLSeq2Seq#8]
- line 196, 227, 288

`masked_e = nn_ops.softmax(e_not_masked) * enc_padding_mask # (batch_size, max_enc_steps)`
`masked_e = nn_ops.softmax(e_not_masked) * enc_padding_mask # (batch_size, max_enc_steps)`
`masked_e = nn_ops.softmax(e_not_masked) * dec_padding_mask[:,:len_dec_states] # (batch_size,len(decoder_states))`

accrding to original paper, should "nn_ops.softmax(e_not_masked)" be "tf.exp(e_not_masked)"   ?

---------------------------------------------------------------------------

- line 289-292

`if len_dec_states <= 1:
            masked_e = array_ops.ones([batch_size,1]) # first step is filled with equal values
            masked_sums = tf.reshape(tf.reduce_sum(masked_e,axis=1),[-1,1]) # (batch_size,1), # if it's zero due to masking we set it to a small value
          decoder_attn_dist = masked_e / masked_sums # (batch_size,len(decoder_states))`

line 291, should  "masked_sums = tf.reshape..." be unindented to the line 291 level?

thanks!
